server/src/main/java/com/thoughtworks/go/server/service/AutoBuild.java

```diff
@@ -88,6 +89,10 @@ public void canProduce(PipelineConfig pipelineConfig, SchedulingCheckerService s
     public boolean isValidBuildCause(PipelineConfig pipelineConfig, BuildCause buildCause) {
         for (MaterialRevision materialRevision : buildCause.getMaterialRevisions()) {
             if (materialRevision.isChanged()) {
+                if (materialRevision.getMaterial() instanceof DependencyMaterial) {
+                    boolean skipSchedulingOnChange = ((DependencyMaterial) (materialRevision.getMaterial())).skipSchedulingOnChange();
+                    return !skipSchedulingOnChange;
```

Comment:
This is just a drive-by comment. Suppose there are two dependency materials which are considered changed (a simple diamond dependency will do), and if the first one is set to skip scheduling on change and the second one is set to not skip scheduling on change, will this work?

Reply:
Yeah, Mahesh pointed that out yesterday. I'll be changing the logic and adding tests to verify it.